src/consts.rs

```diff
@@ -18,24 +18,21 @@ use crate::utils::center_offset;
 /// should less than [`u32::MAX`] as [`Size`] uses [`u32`].
 ///
 /// [const_generics_rfc]: https://rust-lang.github.io/rfcs/2000-const-generics.html
-pub struct CCanvas<C: PixelColor, const W: usize, const H: usize> {
+pub struct CCanvas<C, const W: usize, const H: usize> {
     // we also store the size for working with embedded-graphics
     pub size: Size,
     pub pixels: [[Option<C>; H]; W],
 }
 
-impl<C, const W: usize, const H: usize> Default for CCanvas<C, W, H>
-where
-    C: PixelColor,
-{
+impl<C: Copy + PartialEq, const W: usize, const H: usize> Default for CCanvas<C, W, H> {
```

Comment:
Is Copy and PartialEq necessary here?
I understand why we would need them. Maybe we can replace `Self::new()` as it's the fn adding these trait bounds in this case, or at least that impl.

Reply:
I remember it caused an error or some inconsistency. The issue is that PixelColor does imply Copy and PartialEq, and I remember Copy was required, so I thought I could put both. But I don't remember exactly how I decided that trade off.